Dockerfile

```diff
@@ -1,32 +1,62 @@
 #
```

Comment:
So, I need to look at this a bit more to understand if what I'm going to point out is just nonsensical for our use-case:

in the language-examples of onbuild Dockerfiles that I looked at, they use the non-onbuild Docker image as the base for the onbuild Docker image.  Maybe that doesn't make sense for us??

Ruby:
* https://github.com/docker-library/ruby/blob/master/2.1/Dockerfile
* https://github.com/docker-library/ruby/blob/master/2.1/onbuild/Dockerfile

Go-lang:
* https://github.com/docker-library/golang/blob/1eab0db63794152b4516dbcb70270eb9dced4cbd/1.5/Dockerfile
* https://github.com/docker-library/golang/blob/f1f65c0ab0097a5e3d079d5a74e2468e8d47563d/1.5/onbuild/Dockerfile

So could it make sense for us to follow a similar pattern?  Maybe the configurability of all the various fundamental components (mesos, storm, java) makes this an impossible idea?